app/uk/gov/hmrc/transitmovementsrouter/controllers/MessagesController.scala

```diff
@@ -135,19 +138,36 @@ class MessagesController @Inject() (
     }
   }
 
+  private def extractAuditMessageType(messageType: MessageType): EitherT[Future, MessageTypeExtractionError, AuditType] = {
+    val auditType: EitherT[Future, MessageTypeExtractionError, AuditType] =
+      EitherT.fromOption[Future](messageType.auditType, MessageTypeExtractionError.InvalidMessageType(s"$messageType is not a ResponseMessageType"))
+    auditType
+  }
+
   def incomingViaEIS(ids: ConversationId): Action[Source[ByteString, _]] =
     authenticateEISToken.stream(transformer = eisMessageTransformers.unwrap) {
-      implicit request => _ =>
+      implicit request => size =>
         import MessagesController.PresentationEitherTHelper
 
         val (movementId, triggerId) = ids.toMovementAndMessageId
-
         (for {
           messageType <- messageTypeExtractor.extract(request.headers, request.body).asPresentationWithMessageType(None)
```

Comment:
Here with this call we are not validating about request and response type messages. So here first we need to make sure it is response type message then we need to calculate the audittype. As for request type message we are already doing auditing at api side so need to do auditing for request type message.

Reply:
If EIS sends us a request message, we have bigger problems. If we want to do that validation, that's a tech debt ticket, but I don't really care either way.

Reply:
...though that's done in the extractAuditMessageType function anyway?

Reply:
We should NOT get a request message, otherwise we are in trouble! That's why auditType is set to None as default for request - we then get protection from the call to extractAuditMessage.